pkg/kubelet/cm/container_manager.go

```diff
@@ -91,6 +92,8 @@ type ContainerManager interface {
 	UpdatePluginResources(*schedulercache.NodeInfo, *lifecycle.PodAdmitAttributes) error
 
 	InternalContainerLifecycle() InternalContainerLifecycle
+
+	GetPluginRegistrationHandlerCbkFunc() pluginwatcher.RegisterCallbackFn
```

Comment:
I think returning a Channel rather than asking for a Callback would probably have less impact on this interface and be a more idiomatic model

Reply:
@RenaudWasTaken could you elaborate on why using channel could have less impact on the interface? It seems either way, we either need to pass pluginwatcher to cm.devicemanager or get that information from cm.devicemanager to pluginwatcher by adding an interface func. Regarding these two options, I personally prefer to passing pluginwatcher to cm.devicemanager as how to set up pluginwatcher seems local logic to cm.devicemanager.

Reply:
Because you wouldn't be changing the interface (and thus all the underlying implementations) but just adding an argument to NewLinuxContainerManager().

Reply:
Also prefer to passing pluginwatcher as an argument to NewContainerManager() instead of adding GetPluginRegistrationHandlerCbkFunc to the ContainerManager interface. Looking at NewContainerManager() implementation, I think we may remove devicePluginEnabled arg at the same time and get the flag setting from utilfeature.DefaultFeatureGate, similar to how CPUManager does.